gcc-7 -Wstrict-overflow in SolidMechanicsLagrangianFEM

**Describe the bug**
```
/home/klevtsov/work/GEOSX/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp: In member function ‘virtual geosx::real64 geosx::SolidMechanicsLagrangianFEM::SolverStep(const real64&, const real64&, int, geosx::DomainPartition*)’:
/home/klevtsov/work/GEOSX/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp:308:41: error: assuming signed overflow does not occur when assuming that (X + c) >= X is always true [-Werror=strict-overflow]
     for( integer solveIter=0 ; solveIter<m_maxNumResolves ; ++solveIter )
```

Relevant code:
```c++
    for( integer solveIter=0 ; solveIter<m_maxNumResolves ; ++solveIter )
    {
      dtReturn = NonlinearImplicitStep( time_n, dt, cycleNumber, domain->group_cast<DomainPartition*>(),
                                        getLinearSystemRepository() );
      if( surfaceGenerator!=nullptr )
      {
        if( !( surfaceGenerator->SolverStep( time_n, dt, cycleNumber, domain ) > 0 ) )
        {
          solveIter=m_maxNumResolves;
        }
      }
      else
      {
        solveIter=m_maxNumResolves;
      }
    }
```

The code itself is fine, but the compiler warns that if somehow `m_maxNumResolves == std::numeric_limits<integer>::max()`, the loop increment could overflow the value of `solveIter`, but gcc optimized the code assuming this never happens. It's a pretty contrived and useless warning (in this case). Travis probably didn't catch it because it's building in debug (this warning depends on optimization level).

I think in this case a simple `break` would be fine though?

**To Reproduce**
Steps to reproduce the behavior:
1. Checkout latest `develop`
2. Build Release with gcc-7.3
3. See error

**Expected behavior**
Should compile

**Platform (please complete the following information):**
- Ubuntu 18.04

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,6 +18,7 @@ matrix:
     - MPIFC=mpifort
     - MPIEXEC=/usr/local/bin/mpirun
     - GEOSX_TPL_DIR=/Users/travis/build/GEOSX/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Release
   - compiler: clang6-LC-toss3
     sudo: required
     services: docker
@@ -25,6 +26,7 @@ matrix:
     env:
     - IMG=clang6-LC-toss3
     - GEOSX_TPL_DIR=/home/geosx/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Release
   - compiler: clang7-LC-toss3
     sudo: required
     services: docker
@@ -32,6 +34,15 @@ matrix:
     env:
     - IMG=clang7-LC-toss3
     - GEOSX_TPL_DIR=/home/geosx/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Release
+  - compiler: clang7-LC-toss3-debug
+    sudo: required
+    services: docker
+    dist: trusty
+    env:
+    - IMG=clang7-LC-toss3
+    - GEOSX_TPL_DIR=/home/geosx/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Debug
   - compiler: gcc7-ubuntu18
     sudo: required
     services: docker
@@ -39,6 +50,7 @@ matrix:
     env:
     - IMG=gcc7-ubuntu18
     - GEOSX_TPL_DIR=/home/geosx/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Release
   - compiler: gcc8-ubuntu18
     sudo: required
     services: docker
@@ -46,6 +58,15 @@ matrix:
     env:
     - IMG=gcc8-ubuntu18
     - GEOSX_TPL_DIR=/home/geosx/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Release
+  - compiler: gcc8-ubuntu18-debug
+    sudo: required
+    services: docker
+    dist: trusty
+    env:
+    - IMG=gcc8-ubuntu18
+    - GEOSX_TPL_DIR=/home/geosx/thirdPartyLibs/install-default-release
+    - CMAKE_BUILD_TYPE=Debug
 script: 
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
       pwd;
@@ -59,7 +80,7 @@ script:
       bash ${TRAVIS_BUILD_DIR}/scripts/travis_build_and_test.sh;
     else
       docker run --rm --user='root' -v ${TRAVIS_BUILD_DIR}:/home/geosx/geosx_repo geosx/compiler:$IMG  chown -R geosx /home/geosx;
-      docker run --rm -v ${TRAVIS_BUILD_DIR}:/home/geosx/geosx_repo  -e DO_BUILD -e DO_TEST -e CMAKE_EXTRA_FLAGS -e PATHMOD  -e GEOSX_TPL_DIR geosx/compiler:$IMG  /home/geosx/geosx_repo/scripts/travis_build_and_test.sh;    
+      docker run --rm -v ${TRAVIS_BUILD_DIR}:/home/geosx/geosx_repo  -e DO_BUILD -e DO_TEST -e CMAKE_EXTRA_FLAGS -e PATHMOD  -e GEOSX_TPL_DIR -e CMAKE_BUILD_TYPE geosx/compiler:$IMG  /home/geosx/geosx_repo/scripts/travis_build_and_test.sh;    
     fi
 
 #- docker run --rm --user='root' -v ${TRAVIS_BUILD_DIR}:/home/geosx/geosx_repo geosx/compiler:$IMG  chown -R geosx /home/geosx

--- a/scripts/travis_build_and_test.sh
+++ b/scripts/travis_build_and_test.sh
@@ -24,6 +24,7 @@ if [[ "$DO_BUILD" == "yes" ]] ; then
            -DENABLE_MPI=ON -DMPI_C_COMPILER=${MPICC} -DMPI_CXX_COMPILER=${MPICXX} -DMPI_Fortran_COMPILER=${MPIFC} -DMPIEXEC=${MPIEXEC} -DMPIEXEC_EXECUTABLE=${MPIEXEC} \
            -DGEOSX_TPL_DIR=${GEOSX_TPL_DIR} \
            -DENABLE_SPHINX=OFF \
+           -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
            ${CMAKE_EXTRA_FLAGS} ../src
 
 #    if [[ ${CMAKE_EXTRA_FLAGS} == *COVERAGE* ]] ; then

--- a/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp
+++ b/src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp
@@ -305,7 +305,9 @@ real64 SolidMechanicsLagrangianFEM::SolverStep( real64 const& time_n,
            m_timeIntegrationOption == timeIntegrationOption::QuasiStatic )
   {
     ImplicitStepSetup( time_n, dt, domain, getLinearSystemRepository() );
-    for( integer solveIter=0 ; solveIter<m_maxNumResolves ; ++solveIter )
+
+    localIndex const maxNumResolves = m_maxNumResolves;
+    for( integer solveIter=0 ; solveIter<maxNumResolves ; ++solveIter )
     {
       dtReturn = NonlinearImplicitStep( time_n, dt, cycleNumber, domain->group_cast<DomainPartition*>(),
                                         getLinearSystemRepository() );

